pandas/core/strings/object_array.py

```diff
@@ -468,7 +468,17 @@ def _str_islower(self):
         return self._str_map(str.islower, dtype="bool")
 
     def _str_isnumeric(self):
-        return self._str_map(str.isnumeric, dtype="bool")
+        def is_numeric(value):
+            if value.isnumeric():
+                return True
+            else:
+                try:
+                    float(value)  # Try converting to float (supports negative and decimals)
+                    return True
+                except ValueError:
+                    return False
+
+        return self._str_map(is_numeric, dtype="bool")
```

Comment:
I am negative on this change. As mentioned in the issue, I do not believe we should be changing the behavior of `is_numeric` here.

https://github.com/pandas-dev/pandas/issues/60750#issuecomment-2629402138
https://github.com/pandas-dev/pandas/issues/60750#issuecomment-2608458161

Reply:
I didn't realize this was a PR in your fork - apologies. However, if you do want to contribute to pandas, you need to open up a PR in the pandas repository and not your fork of it.